File: app/src/main/java/org/deiverbum/app/data/entity/LHOfficeBiblicalEntity.java
package org.deiverbum.app.data.entity;

import static org.deiverbum.app.utils.Constants.LH_OFFICE_BIBLICAL;

import androidx.annotation.NonNull;
import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.ForeignKey;

/**
 * @author A. Cedano
 * @version 1.0
 * @since 2023.1
 */

@Entity(tableName = LH_OFFICE_BIBLICAL,
        primaryKeys = {"groupFK", "theOrder"},

        foreignKeys =
                {
                        @ForeignKey(
                                entity = LHOfficeBiblicalJoinEntity.class,
                                parentColumns = "groupID",
                                childColumns = "groupFK",
                                onDelete = ForeignKey.CASCADE,
                                onUpdate = ForeignKey.CASCADE),
                        @ForeignKey(
                                entity = BibleReadingEntity.class,
                                parentColumns = "readingID",
                                childColumns = "readingFK",
                                onDelete = ForeignKey.CASCADE,
                                onUpdate = ForeignKey.CASCADE),
                        @ForeignKey(
                                entity = LHResponsoryEntity.class,
                                parentColumns = "responsoryID",
                                childColumns = "responsoryFK",
                                onDelete = ForeignKey.CASCADE,
                                onUpdate = ForeignKey.CASCADE)
                }
)

public class LHOfficeBiblicalEntity {
    @NonNull
    @ColumnInfo(name = "groupFK")
    public Integer groupFK = 0;

    @NonNull
    @ColumnInfo(name = "readingFK", index = true)
    public Integer readingFK = 0;

    @NonNull
    @ColumnInfo(name = "responsoryFK", index = true)
    public Integer responsoryFK = 0;

    @NonNull
    @ColumnInfo(name = "theme")
    public String theme = "";

    @NonNull
    @ColumnInfo(name = "theOrder", defaultValue = "1")
    public Integer theOrder = 1;
}

File: app/src/main/java/org/deiverbum/app/data/entity/LHOfficeBiblicalEntity.kt
package org.deiverbum.app.data.entity

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.ForeignKey.Companion.CASCADE
import org.deiverbum.app.util.Constants

/**
 * Entidad para la tabla **`lh_office_biblical`** de la base de datos, que se ocupa de gestionar las lecturas bíblicas del Oficio de Lectura de la Liturgia de las Horas.
 *
 * @author A. Cedano
 * @version 1.0
 * @since 2023.1
 */
@Entity(
    tableName = Constants.LH_OFFICE_BIBLICAL,
    primaryKeys = ["groupFK", "theOrder"],
    foreignKeys = [ForeignKey(
        entity = LHOfficeBiblicalJoinEntity::class,
        parentColumns = arrayOf("groupID"),
        childColumns = arrayOf("groupFK"),
        onDelete = CASCADE,
        onUpdate = CASCADE
    ), ForeignKey(
        entity = BibleReadingEntity::class,
        parentColumns = arrayOf("readingID"),
        childColumns = arrayOf("readingFK"),
        onDelete = CASCADE,
        onUpdate = CASCADE
    ), ForeignKey(
        entity = LHResponsoryEntity::class,
        parentColumns = arrayOf("responsoryID"),
        childColumns = arrayOf("responsoryFK"),
        onDelete = CASCADE,
        onUpdate = CASCADE
    )]
)
data class LHOfficeBiblicalEntity(
    @ColumnInfo(name = "groupFK")
    var groupFK: Int,

    @ColumnInfo(name = "readingFK", index = true)
    var readingFK: Int,

    @ColumnInfo(name = "responsoryFK", index = true)
    var responsoryFK: Int,

    @ColumnInfo(name = "theme")
    var theme: String,

    @ColumnInfo(name = "theOrder", defaultValue = "1")
    var theOrder: Int = 1
)